Source/cursor.cpp

```diff
@@ -216,7 +216,10 @@ void CheckTown()
 				trigflag = true;
 				ClearPanel();
 				strcpy(infostr, _("Town Portal"));
-				strcpy(tempstr, fmt::format(_("from {:s}"), Players[missile._misource]._pName).c_str());
+				if (currlevel == 0)
+					strcpy(tempstr, fmt::format(_("to {:s} from {:s}"), LvlNames[Portals[i].level], Players[missile._misource]._pName).c_str());
+				else
+					strcpy(tempstr, fmt::format(_("from {:s}"), Players[missile._misource]._pName).c_str());

```

Comment:
Portals can also lead to quest levels (see `Portal.setlevel`).

```suggestion
				if (currlevel != 0)
					CopyUtf8(tempstr, fmt::format(_("from {:s}"), Players[missile._misource]._pName), sizeof(infostr));				
				else if (Portals[i].setlevel)
					CopyUtf8(tempstr, fmt::format(_("to {:s} from {:s}"), _(QuestLevelNames[Portals[i].level]), Players[missile._misource]._pName), sizeof(infostr));
				else
					CopyUtf8(tempstr, fmt::format(_("to {:s} from {:s}"), _(LvlNames[Portals[i].level]), Players[missile._misource]._pName), sizeof(infostr));
```